web/src/routes/(user)/search/+page.ts

```diff
@@ -9,23 +16,16 @@ export const load = (async (data) => {
   const term =
     url.searchParams.get(QueryParameter.SEARCH_TERM) || url.searchParams.get(QueryParameter.QUERY) || undefined;
   let results: SearchResponseDto | null = null;
+
   if (term) {
-    let params = {};
-    for (const [key, value] of data.url.searchParams) {
-      params = { ...params, [key]: value };
-    }
-    const response = await search({ ...params });
-    let items: AssetResponseDto[] = (data as unknown as { results: SearchResponseDto }).results?.assets.items;
-    if (items) {
-      items.push(...response.assets.items);
+    const payload = JSON.parse(term) as SmartSearchDto | MetadataSearchDto;
+    searchPayload.set(payload);
+
+    if (payload && 'query' in payload) {
```

Comment:
This should check `payload.query` to handle empty strings

Reply:
Because `query` is not in `MetadataSearchDto` so you cannot use that syntax to check.